cypress/integration/tests/upgrade/create_test_data.ts

```diff
@@ -0,0 +1,148 @@
+
+describe("Creating pre-requisites before an upgrade", { tags: "@upgrade" }, () => {
+    let mavenCredsUsername: CredentialsMaven;
+    let sourceControlUsernameCredentials: CredentialsSourceControlUsername;
+
+    before("Login", function () {
+        // Prevent hook from running, if the tag is excluded from run
+        if (hasToBeSkipped("@upgrade")) return;
+
+        // Perform login
+        login();
+    });
+
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+        preservecookies();
+
+        cy.fixture("application").then(function (appData) {
+            this.appData = appData;
+        });
+        cy.fixture("analysis").then(function (analysisData) {
+            this.analysisData = analysisData;
+        });
+        cy.fixture("upgrade-data").then((upgradeData: UpgradeData) => {
+            this.upgradeData = upgradeData;
+        });
+    });
+
+    it("Creating credentials", { tags: "@upgrade" }, function () {
+        const { sourceControlUsernameCredentialsName, mavenUsernameCredentialName } =
+            this.upgradeData;
+        sourceControlUsernameCredentials = new CredentialsSourceControlUsername(
+            getRandomCredentialsData(CredentialType.sourceControl, UserCredentials.usernamePassword)
+        );
+        sourceControlUsernameCredentials.name = sourceControlUsernameCredentialsName;
+        sourceControlUsernameCredentials.create();
+
+        mavenCredsUsername = new CredentialsMaven(getRandomCredentialsData(CredentialType.maven));
+        mavenCredsUsername.name = mavenUsernameCredentialName;
+        mavenCredsUsername.create();
+    });
+
+    it("Creating Controls instances", { tags: "@upgrade" }, function () {
+        const {
+            jobFunctionName,
+            stakeHolderGroupName,
+            stakeHolderName,
+            businessServiceName,
+            tagTypeName,
+            tagName,
+        } = this.upgradeData;
+        const jobFunction = new Jobfunctions(jobFunctionName);
+        // Defining stakeholder groups
+        const stakeHolderGroup = new Stakeholdergroups(stakeHolderGroupName);
+        // Defining stakeholders
+        const stakeHolder = new Stakeholders("test@gmail.com", stakeHolderName, jobFunctionName, [
+            stakeHolderGroupName,
+        ]);
+        // Defining business Service
+        const businessService = new BusinessServices(businessServiceName);
+        // Defining tag type
+        const tagType = new TagType(tagTypeName, data.getColor(), data.getRandomNumber(5, 15));
+        // Defining tag
+        const tag = new Tag(tagName, tagTypeName);
+
+        jobFunction.create();
+        stakeHolderGroup.create();
+        stakeHolder.create();
+        businessService.create();
+        tagType.create();
+        tag.create();
+    });
+
+    it("Creating source applications", { tags: "@upgrade" }, function () {
```

Comment:
These methods to create source , binary applications and analysis are not required .
We will run source_analysis.test.ts, binary_analysis.test.ts and upload_binary.test.ts

They cover it all